onnxruntime/core/providers/dml/DmlExecutionProvider/src/TensorDesc.cpp

```diff
@@ -315,3 +315,52 @@
     }
     m_bufferTensorDesc.DimensionCount = newDimensionCount;
 }
+
+// Uses dimensionMapping to reorder m_sizes and m_strides to match specific Tensor layout
+void TensorDesc::PermuteDimensions(const std::vector<uint32_t> dimensionMapping, const TensorAxis alignment)
```

Comment:
>std::vector<uint32_t> [](http://example.com/codeflow?start=41&length=21)

Avoid passing parameters by value copy (except in some specific cases) because a no-copy view is much cheaper than a heap allocation. Check what the other functions do like `TensorDesc::SetStrides`.